From c8fea7172b3ef52e9a449eb4f8350ff97c949844 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 19 Sep 2014 12:29:31 -0400
Subject: [acpi] hotplug/pci: Use pci_device_is_present()

Message-id: <1411129776-21656-57-git-send-email-prarit@redhat.com>
Patchwork-id: 96385
O-Subject: [RHEL7.1 PATCH BZ 1141846 56/61] ACPI / hotplug / PCI: Use pci_device_is_present()
Bugzilla: 1141846
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Myron Stowe <mstowe@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1141846

commit b8a62d540240387c7e97a69127cab3ab9cadc70a
Author: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Date:   Tue Mar 4 00:48:57 2014 +0100

    ACPI / hotplug / PCI: Use pci_device_is_present()

    Make the ACPI-based PCI hotplug (ACPIPHP) code use
    pci_device_is_present() for checking if devices are present instead
    of open coding the same thing.

    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
    Reviewed-by: Mika Westerberg <mika.westerberg@linux.intel.com>

Cc: Myron Stowe <mstowe@redhat.com>
Cc: Andy Gospodarek <agospoda@redhat.com>
Cc: Tony Camuso <tcamuso@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/hotplug/acpiphp_glue.c b/drivers/pci/hotplug/acpiphp_glue.c
index 0d46d2f..bccc27e 100644
--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -668,12 +668,9 @@ static void trim_stale_devices(struct pci_dev *dev)
   alive = (ACPI_SUCCESS(status) && device_status_valid(sta))
    || acpiphp_no_hotplug(adev);
  }
- if (!alive) {
-  u32 v;
+ if (!alive)
+  alive = pci_device_is_present(dev);
 
-  /* Check if the device responds. */
-  alive = pci_bus_read_dev_vendor_id(dev->bus, dev->devfn, &v, 0);
- }
  if (!alive) {
   pci_stop_and_remove_bus_device(dev);
   if (adev)
-- 
1.7.1